src/main/java/uk/gov/ons/validation/service/ProcessWranglerQuestionData.java

```diff
@@ -0,0 +1,118 @@
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import uk.gov.ons.validation.common.Constants;
+import uk.gov.ons.validation.entity.QuestionInputData;
+import uk.gov.ons.validation.entity.ValidationConfig;
+import lombok.extern.log4j.Log4j2;
+import uk.gov.ons.validation.entity.WranglerRequest;
+import uk.gov.ons.validation.entity.WranglerResponseData;
+import uk.gov.ons.validation.util.PropertiesUtil;
+
+import java.util.List;
+
+import static java.lang.String.format;
+
+@Log4j2
+public class ProcessWranglerQuestionData {
+
+    private static final String SEND_MESSAGE = "Attempting to invoke %s with the json string %s.";
+    private static final String QUES_DER_MESSAGE = "Match Found for both Question and Derived Question. " +
+            "Question Code is %s Derived Question code is %s Question Code value %s " +
+            "and Derived Question code value is %s.";
+
+    /**
+     * Process Question data containing Question code and their values from Data Preparation Lambda
+     *
+     * @param request WranglerRequest
+     * @throws Exception
+     */
+    public void processQuestionAndDerivedData(WranglerRequest request) throws Exception {
+
+        //1. Building Validation Config
+        ValidationConfig validationData = new ValidationConfig();
+        List<ValidationConfig> configuration = validationData.getValidationConfiguration();
+
+        List<QuestionInputData> responses = request.getResponses();
+        log.info("Request Data {} ", responses);
+
+        if (configuration != null && configuration.size() > 0) {
+            for (ValidationConfig config : configuration) {
+                InvokeConfig invokeConfig = newInvokeConfig();
+                for (QuestionInputData inputData : responses) {
+                    invokeConfig.processQuestionCodeFound(inputData, config.getQuestionCode());
+                    invokeConfig.processDerivedQuestFound(inputData, config.getDerivedQuestionCode());
+                    if (invokeConfig.isQuestionCodeFound() && invokeConfig.isDerivedQuestFound()) {
+                        log.info(format(QUES_DER_MESSAGE, invokeConfig.getFinalQuestCode(),
+                                invokeConfig.getFinalDerivedQuestCode(), invokeConfig.getFinalQuestCodeValue(),
+                                invokeConfig.getFinalDerivedQuestValue()));
+                        //Call External Lambda which performs Validation
+                        callValidationLambda(invokeConfig);
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Call Validation Lambda i.e. VET
+     *
+     * @param config InvokeConfig
+     * @throws JsonProcessingException
+     */
+    private void callValidationLambda(InvokeConfig config) throws JsonProcessingException {
+        try {
+            //Call External Lambda which performs Validation
```

Comment:
Again, I don't think you need this comment. The method name tells us what is happening.

Reply:
I removed it. This method has extra logic and for a new developer it would be easy to know at what point this is called. Java Comment will always highlight and understand the code for any one who looks into the code in the first instance. I feel writing an extra comment will always help.

Reply:
Generally if comments are required it means the code is not clear enough to understand in the first place. In this case the method name "callValidationLambda" states what is happening in the method so the comment is superfluous.